src/plugins/discover/public/embeddable/initialize_search_embeddable_api.tsx

```diff
@@ -85,14 +85,21 @@ export const initializeSearchEmbeddableApi = async (
   const searchSource$ = new BehaviorSubject<ISearchSource>(searchSource);
   const dataViews = new BehaviorSubject<DataView[] | undefined>(dataView ? [dataView] : undefined);
 
+  const defaultHeaderRowHeight = DEFAULT_HEADER_ROW_HEIGHT_LINES;
+  const defaultRowHeight = discoverServices.uiSettings.get(ROW_HEIGHT_OPTION);
+  const defaultRowsPerPage = getDefaultRowsPerPage(discoverServices.uiSettings);
+  const defaultSampleSize = discoverServices.uiSettings.get(SAMPLE_SIZE_SETTING);
+
   /** This is the state that can be initialized from the saved initial state */
   const columns$ = new BehaviorSubject<string[] | undefined>(initialState.columns);
   const grid$ = new BehaviorSubject<DiscoverGridSettings | undefined>(initialState.grid);
-  const rowHeight$ = new BehaviorSubject<number | undefined>(initialState.rowHeight);
-  const rowsPerPage$ = new BehaviorSubject<number | undefined>(initialState.rowsPerPage);
   const headerRowHeight$ = new BehaviorSubject<number | undefined>(initialState.headerRowHeight);
-  const sort$ = new BehaviorSubject<SortOrder[] | undefined>(initialState.sort);
+  const rowHeight$ = new BehaviorSubject<number | undefined>(initialState.rowHeight);
+  const rowsPerPage$ = new BehaviorSubject<number | undefined>(
+    initialState.rowsPerPage ?? defaultRowsPerPage
+  );
```

Comment:
@Heenawter Is it a right way of fixing or it should be changed somewhere else?

Reply:
This definitely works!! Did a quick test and, because of how we are serializing this value, `rowsPerPage` will be always saved as part of the panel when it is overwritten by `defaultRowsPerPage` unless `defaultRowsPerPage` is equal to the value of the saved search rows per page. Is this desired? if not, I think you have two options:
1. You leave this as-is and add a special case for `rowPerPage` in your serialize so that you **only add it** to the serialized panel if it's not equal to the saved search `rowsPerPage` **or** the default rows per page.
2. You overwrite in the component instead, so that this value never gets saved.

Reply:
@Heenawter Thanks for the suggestions! I will look then into switching to (2).